scripts/build-controller-release.sh

```diff
@@ -131,7 +136,28 @@ if [[ ! -d $SERVICE_CONTROLLER_SOURCE_PATH ]]; then
     exit 1
 fi
 
-RELEASE_VERSION="$2"
+if [[ $# -eq 2 ]]; then
+  RELEASE_VERSION="$2"
+  # validate that release version is in the format vx.y.z , where x,y,z are
+  # positive real numbers
+  if ! (echo "$RELEASE_VERSION" | grep -Eq "^v[0-9]+\.[0-9]+\.[0-9]+$"); then
+    echo "Release version should have following regex format: ^v[0-9]+\.[0-9]+\.[0-9]+$"
+    exit 1
+  fi
+else
+  # If the release version is not provided, use "v0.0.0-non-release-version".
+  # This non-release version will allow generation of release artifacts and
+  # executing presubmit 'release-test' job on those artifacts.
+  # ACK postsubmit release job makes sure this version does not get released to
+  # public ecr repository.
+  #
+  # Using a static non-release version works because this is only a placeholder
+  # value which gets replaced during presubmit 'release-test' job. Having a
+  # default non-release value also helps AWS service teams to develop the
+  # controller without worrying about the version until actual controller
+  # release.
+  RELEASE_VERSION="v0.0.0-non-release-version"
```

Comment:
We can put the above into the prow job that executes `build-controller-release.sh`, no?

Reply:
None of the presubmit prowjobs execute this script.

Currently all presubmit tests run on artifacts already present in the PR. They are not generating it for service teams.